// @po-ui/ng-components/esm2020/lib/components/po-field/po-field.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { HttpClientModule } from '@angular/common/http';
import { PoButtonGroupModule } from '../po-button-group/index';
import { PoButtonModule } from '../po-button/index';
import { PoCheckboxGroupModule } from './po-checkbox-group/po-checkbox-group.module';
import { PoContainerModule } from '../po-container/index';
import { PoCalendarModule } from '../po-calendar/po-calendar.module';
import { PoCleanModule } from './po-clean/po-clean.module';
import { PoDatepickerModule } from './po-datepicker/po-datepicker.module';
import { PoDisclaimerGroupModule } from './../po-disclaimer-group/po-disclaimer-group.module';
import { PoDisclaimerModule } from './../po-disclaimer/po-disclaimer.module';
import { PoFieldContainerModule } from './po-field-container/po-field-container.module';
import { PoLoadingModule } from '../po-loading/index';
import { PoModalModule } from '../po-modal/po-modal.module';
import { PoProgressModule } from './../po-progress/po-progress.module';
import { PoServicesModule } from '../../services/services.module';
import { PoTableModule } from '../po-table/po-table.module';
import { PoTooltipModule } from './../../directives/po-tooltip/po-tooltip.module';
import { PoIconModule } from '../po-icon/po-icon.module';
import { PoComboComponent } from './po-combo/po-combo.component';
import { PoComboOptionTemplateDirective } from './po-combo/po-combo-option-template/po-combo-option-template.directive';
import { PoDatepickerRangeComponent } from './po-datepicker-range/po-datepicker-range.component';
import { PoDecimalComponent } from './po-decimal/po-decimal.component';
import { PoEmailComponent } from './po-email/po-email.component';
import { PoLoginComponent } from './po-login/po-login.component';
import { PoLookupComponent } from './po-lookup/po-lookup.component';
import { PoLookupModalComponent } from './po-lookup/po-lookup-modal/po-lookup-modal.component';
import { PoMultiselectDropdownComponent } from './po-multiselect/po-multiselect-dropdown/po-multiselect-dropdown.component';
import { PoMultiselectComponent } from './po-multiselect/po-multiselect.component';
import { PoMultiselectItemComponent } from './po-multiselect/po-multiselect-item/po-multiselect-item.component';
import { PoMultiselectSearchComponent } from './po-multiselect/po-multiselect-search/po-multiselect-search.component';
import { PoRichTextBodyComponent } from './po-rich-text/po-rich-text-body/po-rich-text-body.component';
import { PoRichTextComponent } from './po-rich-text/po-rich-text.component';
import { PoRichTextImageModalComponent } from './po-rich-text/po-rich-text-image-modal/po-rich-text-image-modal.component';
import { PoRichTextLinkModalComponent } from './po-rich-text/po-rich-text-link-modal/po-rich-text-link-modal.component';
import { PoRichTextToolbarComponent } from './po-rich-text/po-rich-text-toolbar/po-rich-text-toolbar.component';
import { PoInputComponent } from './po-input/po-input.component';
import { PoNumberComponent } from './po-number/po-number.component';
import { PoPasswordComponent } from './po-password/po-password.component';
import { PoRadioGroupComponent } from './po-radio-group/po-radio-group.component';
import { PoSelectComponent } from './po-select/po-select.component';
import { PoSwitchComponent } from './po-switch/po-switch.component';
import { PoTextareaComponent } from './po-textarea/po-textarea.component';
import { PoUploadComponent } from './po-upload/po-upload.component';
import { PoUploadDragDropComponent } from './po-upload/po-upload-drag-drop/po-upload-drag-drop.component';
import { PoUploadDragDropDirective } from './po-upload/po-upload-drag-drop/po-upload-drag-drop.directive';
import { PoUploadDragDropAreaOverlayComponent } from './po-upload/po-upload-drag-drop/po-upload-drag-drop-area-overlay/po-upload-drag-drop-area-overlay.component';
import { PoUploadDragDropAreaComponent } from './po-upload/po-upload-drag-drop/po-upload-drag-drop-area/po-upload-drag-drop-area.component';
import { PoUploadFileRestrictionsComponent } from './po-upload/po-upload-file-restrictions/po-upload-file-restrictions.component';
import { PoUrlComponent } from './po-url/po-url.component';
import { PoCheckboxModule } from './po-checkbox/po-checkbox.module';
import { PoRadioComponent } from './po-radio/po-radio.component';
import { PoLabelModule } from '../po-label';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "../po-disclaimer/po-disclaimer.component";
import * as i3 from "./po-field-container/po-field-container-bottom/po-field-container-bottom.component";
import * as i4 from "./po-field-container/po-field-container.component";
import * as i5 from "../po-loading/po-loading.component";
/**
 * @description
 *
 * Módulo dos componentes: po-checkbox, po-checkbox-group, po-combo, po-datepicker, po-datepicker-range, po-email, po-input,
 * po-lookup, po-number, po-multiselect, po-password, po-radio-group, po-select, po-switch, po-textarea, po-upload
 * e po-url.
 *
 * > Não esqueça de importar o módulo `FormsModule` para usar os componentes de formulários e caso esteja trabalhando com
 * > formulários reativos, importe o módulo `ReactiveFormsModule`, ambos nativos do Angular.
 */
export class PoFieldModule {
}
PoFieldModule.ɵfac = function PoFieldModule_Factory(t) { return new (t || PoFieldModule)(); };
PoFieldModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoFieldModule });
PoFieldModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [CommonModule,
        FormsModule,
        HttpClientModule,
        PoButtonGroupModule,
        PoButtonModule,
        PoCleanModule,
        PoCalendarModule,
        PoCheckboxGroupModule,
        PoContainerModule,
        PoDatepickerModule,
        PoDisclaimerGroupModule,
        PoDisclaimerModule,
        PoFieldContainerModule,
        PoLoadingModule,
        PoModalModule,
        PoProgressModule,
        PoServicesModule,
        PoTableModule,
        PoTooltipModule,
        PoIconModule,
        PoCheckboxModule,
        PoLabelModule, PoCheckboxGroupModule,
        PoCleanModule,
        PoDatepickerModule,
        PoFieldContainerModule,
        PoCheckboxModule,
        PoLabelModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoFieldModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    FormsModule,
                    HttpClientModule,
                    PoButtonGroupModule,
                    PoButtonModule,
                    PoCleanModule,
                    PoCalendarModule,
                    PoCheckboxGroupModule,
                    PoContainerModule,
                    PoDatepickerModule,
                    PoDisclaimerGroupModule,
                    PoDisclaimerModule,
                    PoFieldContainerModule,
                    PoLoadingModule,
                    PoModalModule,
                    PoProgressModule,
                    PoServicesModule,
                    PoTableModule,
                    PoTooltipModule,
                    PoIconModule,
                    PoCheckboxModule,
                    PoLabelModule
                ],
                exports: [
                    PoCheckboxGroupModule,
                    PoCleanModule,
                    PoDatepickerModule,
                    PoComboComponent,
                    PoComboOptionTemplateDirective,
                    PoDecimalComponent,
                    PoDatepickerRangeComponent,
                    PoEmailComponent,
                    PoFieldContainerModule,
                    PoInputComponent,
                    PoLoginComponent,
                    PoLookupComponent,
                    PoLookupModalComponent,
                    PoMultiselectComponent,
                    PoNumberComponent,
                    PoPasswordComponent,
                    PoRadioGroupComponent,
                    PoRichTextComponent,
                    PoSelectComponent,
                    PoSwitchComponent,
                    PoTextareaComponent,
                    PoUploadComponent,
                    PoUrlComponent,
                    PoRadioComponent,
                    PoCheckboxModule,
                    PoLabelModule
                ],
                declarations: [
                    PoComboComponent,
                    PoComboOptionTemplateDirective,
                    PoDecimalComponent,
                    PoDatepickerRangeComponent,
                    PoEmailComponent,
                    PoInputComponent,
                    PoLoginComponent,
                    PoLookupComponent,
                    PoLookupModalComponent,
                    PoMultiselectComponent,
                    PoMultiselectDropdownComponent,
                    PoMultiselectItemComponent,
                    PoMultiselectSearchComponent,
                    PoNumberComponent,
                    PoPasswordComponent,
                    PoRadioGroupComponent,
                    PoRichTextBodyComponent,
                    PoRichTextComponent,
                    PoRichTextImageModalComponent,
                    PoRichTextLinkModalComponent,
                    PoRichTextToolbarComponent,
                    PoSelectComponent,
                    PoSwitchComponent,
                    PoTextareaComponent,
                    PoUploadComponent,
                    PoUploadDragDropComponent,
                    PoUploadDragDropDirective,
                    PoUploadDragDropAreaOverlayComponent,
                    PoUploadDragDropAreaComponent,
                    PoUploadFileRestrictionsComponent,
                    PoUrlComponent,
                    PoRadioComponent
                ],
                providers: []
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoFieldModule, { declarations: [PoComboComponent,
        PoComboOptionTemplateDirective,
        PoDecimalComponent,
        PoDatepickerRangeComponent,
        PoEmailComponent,
        PoInputComponent,
        PoLoginComponent,
        PoLookupComponent,
        PoLookupModalComponent,
        PoMultiselectComponent,
        PoMultiselectDropdownComponent,
        PoMultiselectItemComponent,
        PoMultiselectSearchComponent,
        PoNumberComponent,
        PoPasswordComponent,
        PoRadioGroupComponent,
        PoRichTextBodyComponent,
        PoRichTextComponent,
        PoRichTextImageModalComponent,
        PoRichTextLinkModalComponent,
        PoRichTextToolbarComponent,
        PoSelectComponent,
        PoSwitchComponent,
        PoTextareaComponent,
        PoUploadComponent,
        PoUploadDragDropComponent,
        PoUploadDragDropDirective,
        PoUploadDragDropAreaOverlayComponent,
        PoUploadDragDropAreaComponent,
        PoUploadFileRestrictionsComponent,
        PoUrlComponent,
        PoRadioComponent], imports: [CommonModule,
        FormsModule,
        HttpClientModule,
        PoButtonGroupModule,
        PoButtonModule,
        PoCleanModule,
        PoCalendarModule,
        PoCheckboxGroupModule,
        PoContainerModule,
        PoDatepickerModule,
        PoDisclaimerGroupModule,
        PoDisclaimerModule,
        PoFieldContainerModule,
        PoLoadingModule,
        PoModalModule,
        PoProgressModule,
        PoServicesModule,
        PoTableModule,
        PoTooltipModule,
        PoIconModule,
        PoCheckboxModule,
        PoLabelModule], exports: [PoCheckboxGroupModule,
        PoCleanModule,
        PoDatepickerModule,
        PoComboComponent,
        PoComboOptionTemplateDirective,
        PoDecimalComponent,
        PoDatepickerRangeComponent,
        PoEmailComponent,
        PoFieldContainerModule,
        PoInputComponent,
        PoLoginComponent,
        PoLookupComponent,
        PoLookupModalComponent,
        PoMultiselectComponent,
        PoNumberComponent,
        PoPasswordComponent,
        PoRadioGroupComponent,
        PoRichTextComponent,
        PoSelectComponent,
        PoSwitchComponent,
        PoTextareaComponent,
        PoUploadComponent,
        PoUrlComponent,
        PoRadioComponent,
        PoCheckboxModule,
        PoLabelModule] }); })();
i0.ɵɵsetComponentScope(PoMultiselectComponent, [i1.NgClass, i1.NgForOf, i1.NgIf, i2.PoDisclaimerComponent, i3.PoFieldContainerBottomComponent, i4.PoFieldContainerComponent, PoMultiselectDropdownComponent], []);
i0.ɵɵsetComponentScope(PoMultiselectDropdownComponent, [i1.NgForOf, i1.NgIf, i5.PoLoadingComponent, PoMultiselectItemComponent,
    PoMultiselectSearchComponent], []);
